flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -1775,8 +1783,39 @@ public ResolvedCatalogView resolveCatalogView(CatalogView view) {
         if (view instanceof ResolvedCatalogView) {
             return (ResolvedCatalogView) view;
         }
+
+        if (view instanceof QueryOperationCatalogView) {
+            return new ResolvedCatalogView(
+                    view, view.getUnresolvedSchema().resolve(schemaResolver));
+        }
+
+        final List<Operation> parse = parser.parse(view.getExpandedQuery());
         final ResolvedSchema resolvedSchema = view.getUnresolvedSchema().resolve(schemaResolver);
-        return new ResolvedCatalogView(view, resolvedSchema);
+        if (parse.size() != 1 || !(parse.get(0) instanceof QueryOperation)) {
+            return new ResolvedCatalogView(view, resolvedSchema);
```

Comment:
in this case you want to let the planner fail, right? we should add a comment here what is the intention of this logic